### PR TITLE
"Perfect error handling and recovery"

### PR DESCRIPTION
**Pull request by taoeffect**
## Changes

- Core changes are in file `state.js` (click the link to make GitHub show the diff!)
- Error generation / testing page at http://localhost:8000/app/error-testing
- Miscellaneous bugfixes and improvements

## Background Reading

- https://github.com/okTurtles/group-income-simple/issues/602
- https://github.com/okTurtles/group-income-simple/issues/610
- https://gitlab.okturtles.org/okturtles/group-income-simple/snippets/9

**Be sure to squash-merge this one.**

### CODE REVIEW

**Review by pieer — APPROVED** [review]

File: frontend/model/state.js [hunk 1]
```diff
@@ -8,42 +8,53 @@ import Vue from 'vue'
...
   }
 }
 
 sbp('sbp/selectors/register', {
-  // This will build the current contract state from applying all its actions
   'state/latestContractState': async (contractID: string) => {
     let events = await sbp('backend/eventsSince', contractID, contractID)
     events = events.map(e => GIMessage.deserialize(e))
-    const state = {}
+    var state = {}
```
Comment (sandrina-p) [inline frontend/model/state.js:43]: when we need to re-assign a variable use [`let`](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Statements/let).

File: frontend/views/components/PageSection.vue [hunk 2]
```diff
@@ -1,7 +1,6 @@
 <template lang="pug">
 section.card
-  h3.title(v-if='title')
-    i18n {{title}}
```
Comment (sandrina-p) [inline frontend/views/components/PageSection.vue:4]: I believe we should keep i18n here.

**Review by sandrina-p — APPROVED** [review]
@greg
It's a lot of stuff but I didn't spot any major error. Do you plan to still do any of the new TODOS you left or can I squash+merge it?

File: frontend/views/components/PageSection.vue (hunk 2 above)
Comment (pieer) [reply frontend/views/components/PageSection.vue:4]: No, the title should be translated already

**Pull request merged**